### PR TITLE
fix: header responsiveness

### LINKED ISSUE #463: Header part is not consistently responsive to different screen sizes.
**Describe the bug**
When the Zubhub main page is subjected to different screen sizes the UI of the header becomes inconsistent.
Different components like create project and search bar tend to overlap each other.

**To Reproduce**

1. Go to  zubhub main page
2. Resize the screen 
3. Observe the UI in different screen sizes

**Expected behavior**
The header should behave consistently with different screen sizes for a better user experience.

**Screenshots**
Screenshot (106)

Screenshot (105)

Screenshot (104)

Screenshot (103)

Screenshot (102)

### PR DESCRIPTION
**Pull request by Ammy198488**
## Summary

Description 

In this pull request, I implemented a responsive design for the header section of the website. I used media queries to adjust the styling of the header for different screen sizes, making it look clean and organized on all devices. 
Closes #463 

## Screenshots
large

Screenshot 2023-03-14 175514

midi

### CODE REVIEW

**Comment by tuxology** [comment]
@Ammy198488 can you please rebase your branch and resolve merge conflicts?  In addition, the search bar is shown on lower resolutions along with the search icon. @iamatifmoin Can you help @Ammy198488 and do a review of this PR since you've already worked on this?

**Comment by Ammy198488** [comment]
Okay @tuxology 
For the images, I should show for other screen sizes right?

**Comment by iamatifmoin** [comment]
> @Ammy198488 can you please rebase your branch and resolve merge conflicts? In addition, the search bar is shown on lower resolutions along with the search icon. @iamatifmoin Can you help @Ammy198488 and do a review of this PR since you've already worked on this?

Yes sure!

**Comment by iamatifmoin** [comment]
@Ammy198488 
Do you know how to rebase and resolve merge conflicts? 
If not, then give me your Zulip ID, and I'll help you out.

**Comment by Ammy198488** [comment]
> @Ammy198488 Do you know how to rebase and resolve merge conflicts? If not, then give me your Zulip ID, and I'll help you out.

okay @iamatifmoin ID is ammy

File: zubhub_frontend/zubhub/src/assets/js/styles/components/breadCrumb/breadCrumbStyle.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ export const styles = theme => ({
     minHeight: '45px',
     textAlign: 'center',
     display: 'flex',
-    width: '100vw',
+    //width: '100%',
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/components/breadCrumb/breadCrumbStyle.js:7]: remove this line

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 2]
```diff
@@ -65,24 +75,40 @@ const styles = theme => ({
           border: '1px solid rgba(0,0,0,0.2)',
         },
       },
+
+      [theme.breakpoints.down('1315')]: {
+        width: '20em',
+
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js:81]: close the block here. --> `},`

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 3]
```diff
@@ -122,22 +157,31 @@ const styles = theme => ({
...
   },
   toggleSearchFormStyle: {
     color: 'white',
   },
+  //  remove the search toggle icon from screens higher than 894
   addOn894: {
+
+    [theme.breakpoints.up('1167')]: {
+
     [theme.breakpoints.up('1168')]: {
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js:156]: remove this line

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 4]
```diff
@@ -122,22 +157,31 @@ const styles = theme => ({
...
       display: 'none',
     },
   },
+  // to remove the search bar from screens lower than 894
   removeOn894: {
+
+    [theme.breakpoints.down('1167')]: {
+
     flex: 1,
     [theme.breakpoints.down('1168')]: {
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js:162]: remove this line

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx [hunk 5]
```diff
@@ -804,6 +804,7 @@ function PageWrapper(props) {
             </ClickAwayListener>
           ) : null}
         </Container>
+        {/* dropdown */}
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: remove this line

File: zubhub_frontend/zubhub/src/assets/js/styles/components/breadCrumb/breadCrumbStyle.js (hunk 1 above)
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/components/breadCrumb/breadCrumbStyle.js]: remove this line

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js (hunk 2 above)
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js]: close the block here --> `},`

**Review by iamatifmoin — CHANGES_REQUESTED** [review]
I have added a few comments. 
Make the changes, and commit the code.
Everything else looks good.

**Comment by Ammy198488** [comment]
@iamatifmoin Thanks
Would look into it

**Comment by Ammy198488** [comment]
@tuxology @iamatifmoin All done and ready.

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 6]
```diff
@@ -40,7 +46,9 @@ const styles = theme => ({
     '& .search-form-input': {
       height: '2.5em',
       maxWidth: '40em',
-      backgroundColor: 'rgba(0,0,0,0.2)',
+      width: '24em',
+      minWidth: 'unset',
+      backgroundColor: 'rgba(255,255,255,0.2)',
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js]: change this line to->

`backgroundColor: 'rgba(0,0,0,0.2)',`

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 7]
```diff
@@ -79,10 +90,12 @@ const styles = theme => ({
     flexFlow: 'row nowrap',
     alignItems: 'center',
     justifyContent: 'center',
+    marginBottom: '1em',
     '& .search-form-input': {
       height: '2.5em',
-      backgroundColor: 'rgba(0,0,0,0.2)',
-      color: 'white',
+      width: '18em',
+      backgroundColor: 'rgba(255,255,255,0.2)',
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js]: change this line to->

`backgroundColor: 'rgba(0,0,0,0.2)',`

File: zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js [hunk 8]
```diff
@@ -79,10 +90,12 @@ const styles = theme => ({
...
+      color: 'black',
```
Comment (iamatifmoin) [inline zubhub_frontend/zubhub/src/assets/js/styles/views/page_wrapper/pageWrapperStyles.js]: change this line to->

`color: 'white',`

**Review by iamatifmoin — CHANGES_REQUESTED** [review]
@Ammy198488 
You have made some changes to the colours that will revert the changes made by #584.
Please make the changes requested in the comments.

**Comment by iamatifmoin** [comment]
Also, is the width of the search bar for this screen-size alright or does it need to be bigger @tuxology?

Zubhub

**Review by iamatifmoin — APPROVED** [review]
@tuxology 
everything looks good.

@Ammy198488 well done!

**Review by tuxology — APPROVED** [review]
LGTM 🌈

**Comment by tuxology** [comment]
@Ammy198488 and @iamatifmoin Actually is it possible to centre the bar or extend the size to remaining width in relation to this: https://github.com/unstructuredstudio/zubhub/pull/652#issuecomment-1469510391 ?

**Comment by iamatifmoin** [comment]
@tuxology 
Maybe you can merge this PR and I can do that in a new PR?
I think that would be better. What do you think?

**Pull request merged**